### PR TITLE
[2.0] Add attribute skipping for attributes that already exist in other forms

### PR DESCRIPTION
**Pull request by strawgate**
Fixes: https://github.com/legrego/homeassistant-elasticsearch/issues/297

### CODE REVIEW

File: custom_components/elasticsearch/es_doc_creator.py [hunk 1]
```diff
@@ -29,6 +29,17 @@
 from custom_components.elasticsearch.system_info import SystemInfo
 
 ALLOWED_ATTRIBUTE_TYPES = tuple | dict | set | list | int | float | bool | str | None
+SKIP_ATTRIBUTES = [
+    "friendly_name",
```
Comment (legrego) [inline custom_components/elasticsearch/es_doc_creator.py:33]: `friendly_name` !== `hass.entity.name` in all cases: https://developers.home-assistant.io/docs/core/entity#entity-naming

File: custom_components/elasticsearch/es_doc_creator.py [hunk 2]
```diff
@@ -29,6 +29,17 @@
...
+    "id",
```
Comment (legrego) [inline custom_components/elasticsearch/es_doc_creator.py:34]: I haven't seen `id` in my attributes, but I'm not opposed to excluding this if you're finding it pervasive and unhelpful.

File: custom_components/elasticsearch/es_doc_creator.py [hunk 3]
```diff
@@ -29,6 +29,17 @@
...
+    "type",
```
Comment (legrego) [inline custom_components/elasticsearch/es_doc_creator.py:35]: What sort of data are you seeing in the `type` attribute? I don't have this one, so I'm not familiar with its contents.

File: custom_components/elasticsearch/es_doc_creator.py [hunk 4]
```diff
@@ -29,6 +29,17 @@
...
+    "entity_picture",
+    "icon",
```
Comment (legrego) [inline custom_components/elasticsearch/es_doc_creator.py:37]: I have no concerns excluding these two, they don't provide any benefit to ES.

File: custom_components/elasticsearch/es_doc_creator.py [hunk 5]
```diff
@@ -29,6 +29,17 @@
...
+    "entity_id",
```
Comment (legrego) [inline custom_components/elasticsearch/es_doc_creator.py:38]: I also haven't seen this attribute, but I'm not opposed to excluding this if you're finding it pervasive and unhelpful.

File: custom_components/elasticsearch/es_doc_creator.py [hunk 6]
```diff
@@ -29,6 +29,17 @@
...
+    "device_class",
+    "state_class",
+    "unit_of_measurement",
```
Comment (legrego) [inline custom_components/elasticsearch/es_doc_creator.py:41]: Where do these fields exist in other forms? I don't see them represented elsewhere in my documents.

Comment (strawgate) [reply custom_components/elasticsearch/es_doc_creator.py:41]: They are added in the floors and labels PR, this PR would merge after floors and labels

File: custom_components/elasticsearch/es_doc_creator.py (hunk 1 above)
Comment (strawgate) [reply custom_components/elasticsearch/es_doc_creator.py:33]: oops i actually thought we already had a field for friendly_name

we could:
1. Add entity.friendly_name
2. update entity.name to follow the logic in the documentation
3. leave it as an attribute that exists on almost every single document

File: custom_components/elasticsearch/es_doc_creator.py (hunk 3 above)
Comment (strawgate) [reply custom_components/elasticsearch/es_doc_creator.py:35]: yeah maybe type should stay

File: custom_components/elasticsearch/es_doc_creator.py (hunk 5 above)
Comment (strawgate) [reply custom_components/elasticsearch/es_doc_creator.py:38]: In almost all cases it just contains the entity id but for `group` sensors and `binary` sensors it appears to be an array of the child entity id's so I will probably re-add it

File: custom_components/elasticsearch/es_doc_creator.py (hunk 2 above)
Comment (strawgate) [reply custom_components/elasticsearch/es_doc_creator.py:34]: i probably jumped the gun on this one too

**Comment by strawgate** [comment]
This is certainly a breaking change that probably should only merge after we create a 1.0 branch

File: custom_components/elasticsearch/es_doc_creator.py (hunk 1 above)
Comment (strawgate) [reply custom_components/elasticsearch/es_doc_creator.py:33]: updated to do 1 but will make this a 2.0 change

**Comment by strawgate** [comment]
@legrego ready to go!

**Comment by strawgate** [comment]
@legrego updated to resolve conflicts

**Review by legrego — APPROVED** [review]

**Pull request merged**